plugin/settings/build.gradle.kts

```diff
@@ -29,9 +29,19 @@ gradlePlugin {
 	}
 }
 
+kotlin {
+	sourceSets.named("main").configure {
+		// Split up the sources into multiple folders, so gradle/plugins can re-use some of them.
+		// Creating an extra folder for only publicly used sources.
+		// Therefore, everything in src/main/kotlin is re-used between this project and gradle/plugins.
+		// See gradle/plugins/build.gradle.kts > kotlin.sourceSets for more info.
+		kotlin.srcDir("src/main/kotlin-published")
```

Comment:
consider copy, because IDEA doesn't recognize methods in gradle/plugins due to duplicate content root.